NitroxServer/Serialization/SaveDataUpgrades/SaveDataUpgrade.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+
+namespace NitroxServer.Serialization.Upgrade
+{
+    public abstract class SaveDataUpgrade
+    {
+        public abstract Version TargetVersion { get; }
```

Comment:
This should probably be a range instead of 1 target version so that really old Nitrox versions aren't erroneously getting "upgraded" to an invalid state.

Reply:
It was intended like this so that for each version there is an upgrade file to it from the last one. I think that a range would complicated thinks with this feature carrying on.

Reply:
```cs
                foreach (SaveDataUpgrade upgrade in upgrades)
                {
                    if (upgrade.TargetVersion > saveFileVersion.Version)
                    {
                        SendMessage(args.Sender, $"Applying {upgrade.GetType().Name}");
                        upgrade.UpgradeData(saveDir, fileEnding);
                    }
                }
```
This code applies all upgrades if TargetVersion is higher so if you tried to upgrade from 0.0.1 to 1.5 you can get issues with order of upgrades and just incompatibilities in general. Unless I'm misunderstanding the code.

Reply:
As far as my testing goes Autofac is sorting them alphabetically . I tested it with `Upgrade_1_4_0_0` and `Upgrade_1_3_5_0` while 1.3.5.0 was executed first.